libcloud/dns/drivers/rcodezero.py

```diff
@@ -0,0 +1,506 @@
+from libcloud.common.types import InvalidCredsError, MalformedResponseError
+from libcloud.dns.base import DNSDriver, Zone, Record
+from libcloud.dns.types import ZoneDoesNotExistError, ZoneAlreadyExistsError
+from libcloud.dns.types import Provider, RecordType
+from libcloud.utils.py3 import httplib
+
+__all__ = [
+    'RcodeZeroDNSDriver',
+]
+
+
+class RcodeZeroResponse(JsonResponse):
+
+    def success(self):
+        i = int(self.status)
+        return 200 <= i <= 299
+
+    def parse_error(self):
+        if self.status == httplib.UNAUTHORIZED:
+            raise InvalidCredsError(
+                'Invalid API key. Check https://my.rcodezero.at/enableapi')
+
+        errors = []
+        try:
+            body = self.parse_body()
+        except MalformedResponseError:
+            e = sys.exc_info()[1]
+            body = '%s: %s' % (e.value, e.body)
+        try:
+            errors = [body['message']]
+        except TypeError:
+            return '%s (HTTP Code: %d)' % (body, self.status)
+        except KeyError:
+            pass
+
+        return '%s (HTTP Code: %d)' % (' '.join(errors), self.status)
+
+
+class RcodeZeroConnection(ConnectionKey):
+    responseCls = RcodeZeroResponse
+
+    def add_default_headers(self, headers):
+        headers['Authorization'] = 'Bearer ' + self.key
+        headers['Accept'] = 'application/json'
+        return headers
+
+
+class RcodeZeroDNSDriver(DNSDriver):
+    type = Provider.RCODEZERO
+    name = 'RcodeZero DNS'
+    website = 'https://www.rcodezero.at/'
+    connectionCls = RcodeZeroConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.AFSDB: 'AFSDB',
+        RecordType.ALIAS: 'ALIAS',
+        RecordType.CERT: 'CERT',
+        RecordType.CNAME: 'CNAME',
+        RecordType.DNAME: 'DNAME',
+        RecordType.DNSKEY: 'DNSKEY',
+        RecordType.DS: 'DS',
+        RecordType.HINFO: 'HINFO',
+        RecordType.KEY: 'KEY',
+        RecordType.LOC: 'LOC',
+        RecordType.MX: 'MX',
+        RecordType.NAPTR: 'NAPTR',
+        RecordType.NS: 'NS',
+        RecordType.NSEC: 'NSEC',
+        RecordType.OPENPGPKEY: 'OPENPGPKEY',
+        RecordType.PTR: 'PTR',
+        RecordType.RP: 'RP',
+        RecordType.RRSIG: 'RRSIG',
+        RecordType.SOA: 'SOA',
+        RecordType.SPF: 'SPF',
+        RecordType.SRV: 'SRV',
+        RecordType.SSHFP: 'SSHFP',
+        RecordType.SRV: 'SRV',
+        RecordType.TLSA: 'TLSA',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __init__(self, key, secret=None, secure=True, host='my.rcodezero.at',
+                 port=None, api_version='v1', **kwargs):
+        """
+        :param    key: API token to be used (required)
+        :type     key: ``str``
+
+        :param    secure: Whether to use HTTPS (default) or HTTP.
+        :type     secure: ``bool``
+
+        :param    host: Hostname used for connections.
+                        Default: ``my.rcodezero.at``.
```

Comment:
I would default to None and put `my.rcodezero.at` as a class-attribute, that way it can easily be patched by a test or another consumer.

Reply:
fixed in ef7b7afa9aaca69ea31706fb38b27e8ee4da30be